internal/commands/secrets/update.go

```diff
@@ -0,0 +1,58 @@
+package secrets
+
+import (
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/spf13/pflag"
+)
+
+type CommandUpdate struct {
+	inputs updateInputs
+	realmClient realm.Client
+	output secretOutput
+}
+
+func (cmd *CommandUpdate) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func (cmd *CommandUpdate) Flags (fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+	fs.StringVarP(&cmd.inputs.secret, flagSecret, flagSecretShort, "", updateFlagSecretUsage)
+	fs.StringVarP(&cmd.inputs.name, flagName, flagNameShort, "", updateFlagNameUsage)
+	fs.StringVarP(&cmd.inputs.value, flagValue, flagValueShort, "", updateFlagValueUsage)
+}
+
+func (cmd *CommandUpdate) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+func (cmd *CommandUpdate) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, err := cli.ResolveApp(ui, cmd.realmClient, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	secretList, err := cmd.realmClient.Secrets(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	toUpdate, err := cmd.inputs.resolveSecret(ui, secretList)
+	if err != nil {
+		return err
+	}
+
+	err = cmd.realmClient.UpdateSecret(app.GroupID, app.ID, toUpdate.ID, cmd.inputs.name, cmd.inputs.value)
```

Comment:
since we're only updating a single secret here, we can just `return cmd.realmClient.UpdateSecret(app.GroupID, app.ID, toUpdate.ID, cmd.inputs.name, cmd.inputs.value)`

and in `Feedback`, you already have the details to print the success message (no need to worry about the error message, that is taken care of by returning the `err` here)

Reply:
how do you get the `err` in the `Feedback` function without assigning it to the `cmd.output`? is this something i'll have to do after your refactor?

Reply:
errors returned from the handler are already handled in `command_factory` when we implement the Cobra command's `RunE` hook (see [here](https://github.com/10gen/realm-cli/blob/next/internal/cli/command_factory.go#L114))

the only reason we used `output` structs in the past was when we tried to do a batch of operations, so that even if an error occurred we would still attempt the rest of the specified requests